EIPS/eip-7053.md

```diff
@@ -0,0 +1,122 @@
+---
+eip: 7053
+title: Interoperable Digital Media Indexing
+description: Interoperable Indexing of Digital Media on the EVM-compatible Blockchains.
+author: Bofu Chen (@bafu), Tammy Yang (@tammyyang)
+discussions-to: https://ethereum-magicians.org/t/eip-7053-interoperable-digital-media-indexing/14394
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-05-22
+---
+
+## Abstract
+
+This EIP proposes an interoperable indexing strategy designed to enhance the organization and retrieval of digital media information across multiple smart contracts and EVM-compatible blockchains. This system enhances the traceability and verification of cross-contract and cross-chain data, facilitating a more efficient discovery of storage locations and crucial information related to media assets. The major purpose is to foster an integrated digital media environment on the blockchain.
+
+## Motivation
+
+Given the significant role digital media files play on the Internet, it's crucial to have a robust and efficient method for indexing immutable information. Existing systems encounter challenges due to the absence of a universal, interoperable identifier for digital media content. This leads to fragmentation and complications in retrieving metadata, storage information, or the provenance of specific media assets. The issues become increasingly critical as the volume of digital media continues to expand.
+
+The motivation behind this EIP is to establish a standardized, decentralized, and interoperable approach to index digital media across EVM-compatible networks. By integrating Decentralized Content Identifiers (CIDs), such as IPFS CID, and Commit events, this EIP puts forward a mechanism enabling unique identification and indexing of each digital media file. Moreover, this system suggests a way for users to access a complete history of data associated with digital media assets, from creation to the current status. This full view enhances transparency, thereby providing users with the necessary information for future interactions with digital media.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Content Identifier
+
+Before the indexing process for digital media can begin, it is REQUIRED to generate unique Content Identifiers for each file. This identifier should the same as the Content Identifiers on the decentralized storage, ensuring each identifier provides access to the metadata, media information, and the content file itself. While there's no explicit requirement for the choice of decentralized storage, it is RECOMMENDED to use IPFS Content Identifiers Version 1 (CIDv1) and follow the same generation rule as Kubo, which is the reference implementation for IPFS. The following also use IPFS CIDv1 as sample implementation.
```

Comment:
Mentioning IPFS in the specification itself is _technically_ an external link, which we don't allow. You have three options forward here, all three of which are somewhat unpleasant:

1. Rewrite the relevant parts of the IPFS documentation here, so in essence describe how to build an IPFS CID here in this EIP.
2. Make a pull request under [EIP-5757](https://eips.ethereum.org/EIPS/eip-5757) to change the rules, and allow linking to the IPFS documentation (or commits.)
3. Don't define CID in this document, and leave it implementation defined.

None of those are great options, so I have to apologize for that.